docs/content/utilities/parser.mdx

```diff
@@ -0,0 +1,349 @@
+
+class HelloWorldModel(BaseModel):
+    message: str
+    sender: str
+
+    @validator('*') # highlight-line
+    def has_whitespace(cls, v):
+        if ' ' not in v:
+            raise ValueError("Must have whitespace...")
+
+        return v
+
+parse(model=HelloWorldModel, event={"message": "hello universe", "sender": "universe"})
+```
+
+#### Validating entire model
+
+`root_validator` can help when you have a complex validation mechanism. For example finding whether data has been omitted, comparing field values, etc.
+
+```python:title=validate_all_field_values.py
+from aws_lambda_powertools.utilities.parser import parse, BaseModel, validator
+
+class UserModel(BaseModel):
+    username: str
+    password1: str
+    password2: str
+
+    @root_validator
+    def check_passwords_match(cls, values):
+        pw1, pw2 = values.get('password1'), values.get('password2')
+        if pw1 is not None and pw2 is not None and pw1 != pw2:
+            raise ValueError('passwords do not match')
+        return values
+
+payload = {
+    "username": "universe",
+    "password1": "myp@ssword",
+    "password2": "repeat password"
+}
+
+parse(model=UserModel, event=payload)
+```
+
+<Note type="info">
+    You can read more about validating list items, reusing validators, validating raw inputs, and a lot more in <a href="https://pydantic-docs.helpmanual.io/usage/validators/">Pydantic's documentation</a>.
+</Note><br/>
+
+
+## Envelopes
+
+Envelope parameter is useful when your actual payload is wrapped around a known structure, for example Lambda Event Sources like Amazon EventBridge.
+
+Example of parsing a model found in an event coming from EventBridge, where all you want is what's inside the `detail` key.
+
+```python:title=parse_eventbridge_payload.py
+from aws_lambda_powertools.utilities.parser import parse, BaseModel, envelopes
+
+class UserModel(BaseModel):
+    username: str
+    password1: str
+    password2: str
+
+payload = {
+    "version": "0",
+    "id": "6a7e8feb-b491-4cf7-a9f1-bf3703467718",
+    "detail-type": "CustomerSignedUp",
+    "source": "CustomerService",
+    "account": "111122223333",
+    "time": "2020-10-22T18:43:48Z",
+    "region": "us-west-1",
+    "resources": ["some_additional_"],
+    # highlight-start
+    "detail": {
+        "username": "universe",
+        "password1": "myp@ssword",
+        "password2": "repeat password"
+    }
+    # highlight-end
+}
+
+ret = parse(model=UserModel, envelope=envelopes.EventBridgeModel, payload=payload) # highlight-line
+
+# Parsed model only contains our actual model, not the entire EventBridge + Payload parsed
+assert ret.password1 == ret.password2
+```
+
+**What's going on here, you might ask**:
+
+1. We imported built-in `models` from the parser utility
+2. Used `envelopes.EventBridgeModel` as the envelope for our `UserModel` model
+3. Parser parsed the original event against the EventBridge model
+4  Parser then parsed the `detail` key using `UserModel`
```

Comment:
```suggestion
4. Parser then parsed the `detail` key using `UserModel`
```

Reply:
Thanks as always @Nr18 ;)

Reply:
So did not comment on it because I was not sure but my assumption is that the parser that is used to build the documentation can handle the `3.1. This confirms that data received conforms with how EventBridge wraps events` notation used on line 348?

Because GitHub is rending it incorrect

Reply:
Aha! good catch, I can see that the malformed rendering now - Merged the two in a single sentence now :) Thank you!